pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/argoproj.io/v1alpha1/Workflow/customizations.yaml

```diff
@@ -0,0 +1,449 @@
+            conditionsIndex = conditionsIndex + 1
+            conditions[conditionsIndex] = condition
+          end
+          if #workflowPodRunning > 0 then        
+            condition = {}
+            condition.type = 'PodRunning'
+            condition.status = 'True'
+            condition.message = 'Workflow executed PodRunning in member clusters: '
+            for i = 1, #workflowPodRunning - 1 do
+              condition.message = condition.message..workflowPodRunning[i]..', '
+            end
+            condition.message = condition.message..workflowPodRunning[#workflowPodRunning]
+            conditionsIndex = conditionsIndex + 1
+            conditions[conditionsIndex] = condition
+          end                
+          if #workflowCompleted > 0 then 
+            condition = {}
+            condition.type = 'Completed'
+            condition.status = 'True'
+            condition.message = 'Workflow executed Completed in member clusters: '
+            for i = 1, #workflowCompleted - 1 do
+              condition.message = condition.message..workflowCompleted[i]..', '
+            end
+            condition.message = condition.message..workflowCompleted[#workflowCompleted]
+            conditionsIndex = conditionsIndex + 1
+            conditions[conditionsIndex] = condition 
+          end
+          desiredObj.metadata.generation = generation
+          desiredObj.status.phase = phase
+          desiredObj.status.startedAt = startedAt
+          desiredObj.status.finishedAt = finishedAt
+          desiredObj.status.estimatedDuration = estimatedDuration
+          desiredObj.status.progress = progress
+          desiredObj.status.message = message
+          desiredObj.status.resourcesDuration = resourcesDuration
+          desiredObj.status.conditions = conditions
+          desiredObj.status.compressedNodes = compressedNodes
+          desiredObj.status.nodes = nodes
+          desiredObj.status.offloadNodeStatusVersion = offloadNodeStatusVersion
+          desiredObj.status.storedTemplates = storedTemplates
+          desiredObj.status.persistentVolumeClaims = persistentVolumeClaims
+          desiredObj.status.outputs = outputs
+          desiredObj.status.storedWorkflowTemplateSpec = storedWorkflowTemplateSpec
+          desiredObj.status.synchronization = synchronization
+          desiredObj.status.artifactRepositoryRef = artifactRepositoryRef
+          desiredObj.status.artifactGCStatus = artifactGCStatus
+          return desiredObj
+        end
+    retention:
+      luaScript: >
+        function Retain(desiredObj, observedObj)
+          if observedObj.metadata ~= nil then
+            desiredObj.metadata = observedObj.metadata
+          end
+          if observedObj.spec ~= nil then
+            desiredObj.spec = observedObj.spec
+          end
+          if observedObj.status ~= nil then
+            desiredObj.status = observedObj.status
+          end
```

Comment:
[ref1](https://github.com/karmada-io/karmada/blob/master/pkg/controllers/status/work_status_controller.go#L224) --> [ref2](https://github.com/karmada-io/karmada/blob/master/pkg/util/objectwatcher/objectwatcher.go#L144)
It seems that [desiredObj](https://github.com/karmada-io/karmada/blob/master/pkg/controllers/status/work_status_controller.go#L205) is converted from `workObject.Spec.Workload.Manifests` which is designed to store `.spec` of `resourceTemplate`. Thus `.status` shouldn't be retain. [ref3](https://github.com/karmada-io/karmada/blob/master/pkg/controllers/status/work_status_controller.go#L238) shows that `.status` is updated into `workCopy.status.manifestStatuses`.

The retention rule may be:
```
        function Retain(desiredObj, observedObj)
          if observedObj.spec ~= nil then
            desiredObj.spec = observedObj.spec
          end
          return desiredObj
        end
```

*journal*
I changed it and test, then I found that the scene go back to the beginning. There are someting unexpected in `argo/workflow-controller` of member cluster.